docs/event-sourcing.md

```diff
@@ -0,0 +1,128 @@
+
+* **Barry O Sullivan** makes a good case for the what and why of event-sourcing in his blog post, [Event Sourcing: What it is and why it's awesome](https://dev.to/barryosull/event-sourcing-what-it-is-and-why-its-awesome).
+
+* **Ethan Garofolo** provides a solid walk-through of designing an event-sourced system using Javascript in his book, [_Practical Microservices_](https://pragprog.com/book/egmicro/practical-microservices).
+
+* **Scott Bellware** gives a clear explanation of evented autonomous services as implemented in Eventide, the inspiration behind Hermes, in this conference talk:
+
+<div style="display: flex; justify-content: center;">
+  <iframe width="560" height="315" src="https://www.youtube.com/embed/qgKlu5gFsJM" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
+In addition, the Postgres event store [Message DB](https://github.com/message-db/message-db) which backs Hermes was extracted from [Eventide](https://eventide-project.org/).  The structure and terminology of Hermes were also heavily influenced by the feature-set and implementation of Eventide.  Consequently, the [Eventide documentation](http://docs.eventide-project.org/) contains a great storehouse of knowledge offering greater insight into the underlying mechanics of Hermes.  I recommend that you peruse them as well.
+
+Assuming you've been fully indoctrinated by the materials above, the remainder of this document will serve to explain event-sourcing as modeled in Message DB and Hermes.
+
+## Service
+
+> A **service** is a **process**.
+
+More specifically, services implement business processes.  They _do_ things.  They cannot be asked for information, nor do they ask anything of anyone else.  Consequently services are **autonomous** by nature, acting by themselves, reacting only to messages in the event store.  If one service in your system goes down, the others continue unaffected.
+
+!> **The word "service" is an overloaded term,** so it's important to make it very clear what services are not.  Services are not REST API's, containerized applications, web endpoints, or database models.  While common uses for the term, they do not fit the definition.
+
+Services consume [streams](/event-sourcing?id=stream) of [messages](/event-sourcing?id=message) and then _do_ things based on those messages.  They are built using one or more [consumers](/api?id=consumer).  Services handle messages **idempotently**, so that previously processed messages can be handled again safely.
+
+Services come in two flavors:
+
+- **Components**, which are services that **perform work** initiated by **commands and events**.
+- **Aggregators**, which are services that **update read-models** based on **events**.
+
+[ TODO: include The Diagram™ here ]
```

Comment:
Still todo?

Reply:
Forgot about this one.  You've seen it before:  the cycle of messages and data through an event-sourced system.  My options were to make one or "borrow" one.  I hadn't decided which to do.